### PR TITLE
volnoti: init at git-2019-09-26

### PR DESCRIPTION
**Pull request by gilligan**
###### Motivation for this change

Add volnoti - https://github.com/davidbrazdil/volnoti

---

I am having some troubles getting this to work. When I start volnoti I get:

```
$ volnoti
ERROR: Couldn't load volume_high.svg. (Couldn't recognize the image file format for file '/nix/store/p5ys8c8g0xflhxxyv8f728y6jgbm0nzf-volnoti-git-2019-09-26/share/pixmaps/volnoti/volume_high.svg')
```
I guess this is [this code](https://github.com/davidbrazdil/volnoti/blob/4af7c8e54ecc499097121909f02ecb42a8a60d24/src/daemon.c#L292) so for some reason `gdk_pixbuf_new_from_file` cannot load the `svg` file. Does anyone have any idea on how to fix this?

Thank you!

### CODE REVIEW

**Comment by gilligan** [comment]
@garbas you said you might be interested in this volume app - here it is. No clue how to fix this though ;)

**Comment by rycee** [comment]
Try adding `librsvg` as a build input. I've previously used that with success but mostly for gtk3 applications.

**Comment by gilligan** [comment]
@rycee I tried that but it didn't change anything ;/

**Comment by rycee** [comment]
@gilligan Sorry! Forgot that you also need to add `wrapGAppsHook` to the native build inputs, otherwise the `librsvg` library won't be visible to GTK. I tried building with these two additional dependencies and it seems to start OK and using `volnoti-show` shows the volume display.

File: pkgs/applications/misc/volnoti/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub
+, pkgconfig, dbus, gdk_pixbuf, glib, libX11, gtk2
+, dbus_glib, autoreconfHook }:
+
+stdenv.mkDerivation rec {
+  name = "volnoti-${version}";
+  version = "git-2019-09-26";
```
Comment (rycee) [inline pkgs/applications/misc/volnoti/default.nix]: Please change name to `volnoti-unstable-${version}` and version to `2013-09-23`. See about [package naming](https://nixos.org/nixpkgs/manual/#sec-package-naming) in the Nixpkgs manual.

**Comment by rycee** [comment]
Looks like a cool tool. If you fix the build inputs and the naming then I'd be happy to merge.

**Comment by fpletz** [comment]
You also need `wrapGAppsHook` because `librsvg` provides a gdk-pixbuf module for reading SVGs but gdk-pixbuf needs the svg module path to load it.

Works for me with this patch:

```
diff --git a/pkgs/applications/misc/volnoti/default.nix b/pkgs/applications/misc/volnoti/default.nix
index 869aea1..39dee90 100644
--- a/pkgs/applications/misc/volnoti/default.nix
+++ b/pkgs/applications/misc/volnoti/default.nix
@@ -1,6 +1,6 @@
 { stdenv, fetchFromGitHub
-, pkgconfig, dbus, gdk_pixbuf, glib, libX11, gtk2
-, dbus_glib, autoreconfHook }:
+, pkgconfig, dbus, gdk_pixbuf, glib, libX11, gtk2, librsvg
+, dbus_glib, autoreconfHook, wrapGAppsHook }:
 
 stdenv.mkDerivation rec {
   name = "volnoti-${version}";
@@ -19,10 +19,10 @@ stdenv.mkDerivation rec {
    ./prepare.sh
   '';
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook wrapGAppsHook ];
 
   buildInputs = [
-    pkgconfig dbus gdk_pixbuf glib libX11 gtk2 dbus_glib autoreconfHook
+    dbus gdk_pixbuf glib libX11 gtk2 dbus_glib librsvg
   ];
 
   meta = with stdenv.lib; {
```

**Comment by fpletz** [comment]
Ah, @rycee beat me. :beers: ;)

File: pkgs/applications/misc/volnoti/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "davidbrazdil";
+    repo = "volnoti";
+    rev = "4af7c8e54ecc499097121909f02ecb42a8a60d24";
+    sha256 = "155lb7w563dkdkdn4752hl0zjhgnq3j4cvs9z98nb25k1xpmpki7";
+  };
+
+  patches = [ ./dbus-headers-fix.patch ];
+
+  preConfigure = ''
+   ./prepare.sh
```
Comment (rycee) [inline pkgs/applications/misc/volnoti/default.nix]: This script just runs `autoreconf`, which `autoreconfHook` already takes care of so can remove the `preConfigure` field completely.

**Comment by rycee** [comment]
@fpletz :smile:

**Comment by gilligan** [comment]
Great - thanks for your help @rycee @fpletz ! It is indeed working just fine now.

**Comment by rycee** [comment]
Cool. I'll merge to master.

**Comment by rycee** [comment]
I rebased into master in f49f90bfdd58e0f02d709c028a6e33a588dd56f4 with two changes:

1. the commit message was changed to `volnoti: init at 2013-09-23`;
2. the patch is downloaded during build rather than stored in Nixpkgs.

**Pull request closed without merging**